### ISSUE TITLE
bug: infiniteScroll throws exceptions in some use cases

### ISSUE THREAD

**Issue by realityfilter** [driftyco/ionic#3574]
**Type**: <span ionic-type>bug</span>

**Platform**: <span ionic-platform>all</span>  

<span ionic-description>Line 82 in infiniteScroll directive should handle the absence of the optional scrollCtrl.

```var jsScrolling = infiniteScrollCtrl.jsScrolling = !scrollCtrl.isNative();```
Should be
```var jsScrolling = infiniteScrollCtrl.jsScrolling = scrollCtrl && !scrollCtrl.isNative() || false;```

Line 46-50 in infiniteScroll controller should handle native scrolling correctly.
```
      if (self.jsScrolling) self.scrollView.resize();
      // only check bounds again immediately if the page isn't cached (scroll el has height)
      if (self.scrollView.__container && self.scrollView.__container.offsetHeight > 0) {
        self.checkBounds();
      }
```
Should be:

```
        if (self.jsScrolling) {
            self.scrollView.resize();
            // only check bounds again immediately if the page isn't cached (scroll el has height)
            if (self.scrollView.__container && self.scrollView.__container.offsetHeight > 0) {
                self.checkBounds();
            }
        }
```
</span>

<span is-issue-template></span>

**Comment by Ionitron** [comment]
Greetings @realityfilter!

My sensors indicate that you need to update your issue through our [custom issue form](http://ionicframework.com/submit-issue/#/?iid=3574). We are now requiring all issues to be submitted this way, to ensure that we have all of the information necessary to fix them as quickly as possible.

**[Click Here To Update Your Issue](http://ionicframework.com/submit-issue/#/?iid=3574)**

I will have no choice but to close this issue if it is not resubmitted through the form. Please fill out the rest of the form, so that I may use my friendly robot powers to assist you.

Thank you!

<span ionitron-issue-resubmit></span>

**Comment by mhartington** [comment]
Is there any way you can show us a demo of why this should change? Scrolling in general will be re worked post 1.0. @perrygovier?

**Comment by realityfilter** [comment]
First issue: 

The `scrollCtrl` is optionally required in the directive. So the infinite scroll directive can be used outside `ion-scroll` with normal divs that only have 'overflow-scroll' specified. In this use case `scrollCtrl` is undefined and calling `scrollCtrl.isNative()` throws an error. So either require the controller without specifying it optional with '?' or change line 82 to handle an `undefined` `scrollCtrl`.

Example:
```
<div class="overflow-scroll"><ion-infinite-scroll /></div>
```

Second Issue:

This is a regression that was introduced in 1.0.0-rc-4. 
Consider this use case:
```
<ion-scroll overflow-scroll="true"><ion-infinite-scroll /></ion-scroll>
```
With native scrolling `self.scrollView` is undefined. An error occurs in this case while evaluating the added guard to checkBounds().

**Comment by realityfilter** [comment]
The second issue is a regression introduced in 1.0.0-rc.4 and should be fixed before 1.0.

**Comment by jacargentina** [comment]
I'm having this second error too; i don't know how are PR handled on this project. If i make one, would it be rapidly evaluated/applied ?

Cheers!

**Comment by perrygovier** [comment]
fixed in 86565e2462c5ec6cec49c47170a279419e8fba73
